### DISCUSSION TITLE
Forces and moments

### DISCUSSION THREAD

**Discussion by Chris2814** [JSBSim-Team/jsbsim#527] [Q&A]
Hi,

I am trying to figure out how the forces and moments acting on the vehicle are calculated during simulations. 
My issue is that the result from the functions defined in the aircraft XML file seem to differ from the actual forces and moments used when solving the body accelerations... 

I access the forces and moments acting on the body using the properties "forces/fbx-total-lbs" (same for y and z) and "moments/l-total-lbsft" (same for M and N) and compare these to the functions in the XML file, i.e. "aero/force/Lift_alpha" e.t.c. The functions defined in the XML seem to be in the stability plane, so these need to be converted into the body frame. Doing so gives a slightly different answer for the forces (but still in an acceptable range). For the moments however, the results differ a lot. 

Is someone aware of why, and could possible enlighten me on how the forces and (especially) moments are calculated before they are used for calculation of the body accelerations?

**Comment by seanmcleod** [comment]
The reason you're seeing a difference in terms of the moments you add up and `moments/lmn-total-lbsft` is due to the last term below, which will be non-zero if the cg and AeroRP aren't coincident.

https://github.com/JSBSim-Team/jsbsim/blob/dd97e99032e6cce6ae3d0d91730f3c10f8ce94af/src/models/FGAerodynamics.cpp#L280

**Reply by Chris2814** [reply]
Alright! Thanks for the answer. That makes sense. 

From my aircraft XML file I can read a difference between AERORP and CG. However only in the Z-axis. The difference in the Z-axis is quite small... Smaller than what I would consider reasonable for the huge difference in pitching moment I get when I compare the two alternatives mentioned earlier. Do you happen to know if there is anything more contributing to, at least the pitching moment? Will the thrust from the engine contribute? Reasonably it should as it is mounted (in my aircraft XML) below the Z CG point. 
Also, is this vDXYZcg a constant value or does it change continuously? When looking in the body frame it would be reasonable for it not to change (neglecting changes in CG due to fuel being consumed)... But is that a correct assumption? Is vDXYZcg expressed in the body frame?

**Comment by seanmcleod** [comment]
Yes any thrust from an engine that is above or below the cg will generate a pitching moment.

> Also, is this vDXYZcg a constant value or does it change continuously?

Yes it gets updated every frame. Typically due to burning fuel, but some people may have point masses for bombs, external tanks etc. that they may jettison that will also change the cg.
